automated/utils/upload-to-squad.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/sh
+
+ATTACHMENT=""
+ARTIFACTORIAL_URL=""
+CURL_VERBOSE_FLAG=""
+FAILURE_RETURN_VALUE=0
+
+usage() {
+    echo "Usage: $0 [-a <attachment>] [-u <artifactorial_url>] [-t <artifactorial_token>] [-v] [-r]" 1>&2
+    echo "  -a attachment           Path to the file to upload" 1>&2
+    echo "  -u squad_url            SQUAD_URL where the attachment will be uploaded to" 1>&2
+    echo "                          This script will try to fetch the SQUAD_ARCHIVE_SUBMIT_TOKEN" 1>&2
+    echo "                          token from (lava_test_dir)/secrets or environments for the upload." 1>&2
+    echo "  -v      Pass -v (verbose) flag to curl for debugging." 1>&2
+    echo "  -r      Report failure. If the upload fails and this flag is set, the script will exit" 1>&2
+    echo "          with return value 1. If the upload is skipped (no URL or no token found)," 1>&2
+    echo "          this script will still return 0." 1>&2
+    exit 1
+}
+
+while getopts ":a:u:t:vr" opt; do
+    case "${opt}" in
+        a) ATTACHMENT="${OPTARG}" ;;
+        u) ARTIFACTORIAL_URL="${OPTARG}" ;;
+        v) CURL_VERBOSE_FLAG="-v" ;;
+        r) FAILURE_RETURN_VALUE=1 ;;
+        *) usage ;;
+    esac
+done
+
+if [ -z "${ARTIFACTORIAL_URL}" ]; then
+    echo "test-attachment skip"
+    command -v lava-test-case > /dev/null 2>&1 && lava-test-case "test-attachment" --result "skip"
+    exit 0
+fi
+
+if command -v lava-test-reference > /dev/null 2>&1; then
+    # The 'ARTIFACTORIAL_TOKEN' needs to be defined in 'secrects' dictionary in job
+    # definition file, it will be used.
+    lava_test_dir="$(find /lava-* -maxdepth 0 -type d | grep -E '^/lava-[0-9]+' 2>/dev/null | sort | tail -1)"
+    if test -f "${lava_test_dir}/secrets"; then
+        # shellcheck disable=SC1090
+        . "${lava_test_dir}/secrets"
+    fi
+
+    if [ -z "${SQUAD_ARCHIVE_SUBMIT_TOKEN}" ]; then
+        echo "WARNING: SQUAD_ARCHIVE_SUBMIT_TOKEN is empty! File uploading skipped."
+        echo "test-attachment skip"
+        command -v lava-test-case > /dev/null 2>&1 && lava-test-case "test-attachment" --result "skip"
+        exit 0
+    else
+        # return is the squad testrun id
+        return=$(curl ${CURL_VERBOSE_FLAG} --header "Auth-Token: ${SQUAD_ARCHIVE_SUBMIT_TOKEN}" --form "attachment=@${ATTACHMENT}" "${ARTIFACTORIAL_URL}")
+    fi
+
+    attachmentBasename="$(basename "${ATTACHMENT}")"
+    if echo "${return}" | grep -E "^[0-9]+$"; then
+        url_uploaded="https://qa-reports.linaro.org/api/testruns/${return}/attachments/?filename=${attachmentBasename}"
```

Comment:
shouldn't ARTIFACTORIAL_URL be used instead of hardcoded qa-reports.linaro.org?

Reply:
thanks for the suggestion, will update to use ARTIFACTORIAL_URL to get the new url.